src/query_plan/fetch_dependency_graph.rs

```diff
@@ -174,15 +204,64 @@ pub(crate) struct FetchDependencyGraphNodePath {
 
 #[derive(Debug, Clone)]
 pub(crate) struct DeferContext {
-    current_defer_ref: Option<NodeStr>,
+    current_defer_ref: Option<DeferRef>,
     path_to_defer_parent: Arc<OpPath>,
-    active_defer_ref: Option<NodeStr>,
+    active_defer_ref: Option<DeferRef>,
     is_part_of_query: bool,
 }
 
-struct ParentRelation<'a> {
+/// Used in `FetchDependencyGraph` to store, for a given node, information about one of its parent.
+/// Namely, this structure stores:
+/// 1. the actual parent node index, and
+/// 2. the path of the node for which this is a "parent relation" into said parent (`path_in_parent`). This information
+///    is maintained for the case where we want/need to merge nodes into each other. One can roughly think of
+///    this as similar to a `mergeAt`, but that is relative to the start of `group`. It can be `None`, which
+///    either mean we don't know that path or that this simply doesn't make sense (there is case where a child `mergeAt` can
+///    be shorter than its parent's, in which case the `path`, which is essentially `child-mergeAt - parent-mergeAt`, does
+///    not make sense (or rather, it's negative, which we cannot represent)). Tl;dr, `None` for the `path` means that
+///    should make no assumption and bail on any merging that uses said path.
+// PORT_NOTE: In JS this uses reference equality, not structural equality, so maybe we should just
+// do pointer comparisons?
+#[derive(Debug, Clone, PartialEq)]
+struct ParentRelation {
     parent_node_id: NodeIndex,
-    path_in_parent: Option<&'a Arc<OpPath>>,
+    path_in_parent: Option<Arc<OpPath>>,
```

Comment:
The lifetime `'a` was tied to the fetch dependency graph, so you could never mutate it while a ParentRelation also exists. Having an owned Arc makes the lifetimes independent of each other